src/md5-browser.js

```diff
@@ -155,14 +159,15 @@ function wordsToMd5(x, len) {
  * Characters >255 have their high-byte silently ignored.
  */
 function bytesToWords(input) {
-  var i;
-  var output = [];
-  output[(input.length >> 2) - 1] = undefined;
-  for (i = 0; i < output.length; i += 1) {
-    output[i] = 0;
+  if (input.length === 0) {
+    return [];
   }
-  var length8 = input.length * 8;
-  for (i = 0; i < length8; i += 8) {
+
+  const length8 = input.length * 8;
+
+  const output = new Uint32Array(getOutputLength(length8));
```

Comment:
Why a `Uint32Array` instead of a plain `Array`?

Reply:
https://v8.dev/blog/elements-kinds

Reply:
Should allow for better JS engine optimizations.  'Pretty sure typed arrays are supported on all our supported platforms, so this should be okay.

Reply:
Hmm, what made me suspicious is the following. With this adjustment to the benchmarks:

```diff
--- a/examples/benchmark/benchmark.js
+++ b/examples/benchmark/benchmark.js
@@ -7,7 +7,7 @@ const uuidv5 = (typeof window !== 'undefined' && window.uuidv5) || require('uuid

 console.log('Starting. Tests take ~1 minute to run ...');

-const array = new Array(16);
+const array = new Uint8Array(16);
 const suite = new Benchmark.Suite();
 suite
   .add('uuidv1()', function () {
```

the results change significantly.

With `new Array(16)`:

```
uuidv1() fill existing array x 4,749,694 ops/sec ±3.14% (87 runs sampled)
uuidv4() fill existing array x 356,833 ops/sec ±3.75% (83 runs sampled)
```

With `new Uint8Array(16)`:

```
uuidv1() fill existing array x 775,015 ops/sec ±1.50% (93 runs sampled)
uuidv4() fill existing array x 364,461 ops/sec ±4.00% (85 runs sampled)
```

What am I missing?

Reply:
@ctavan hmm…

On my machine the difference in the margin of error. Try again…
Although, I made some more optimizations locally.